src/services/SignatureHelper.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @copyright 2024 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+
+declare(strict_types=1);
+
+namespace Elabftw\Services;
+
+use DateTimeImmutable;
+use Elabftw\Elabftw\App;
+use Elabftw\Elabftw\MinisignKeys;
+use Elabftw\Enums\Meaning;
+use Elabftw\Models\Config;
+use Elabftw\Models\Users;
+
+use ParagonIE\ConstantTime\Base64;
+
+use function sodium_crypto_generichash;
+
+/**
+ * Helper class for minisign compatible signatures
+ * minisign by Frank Denis: https://jedisct1.github.io/minisign/#secret-key-format
+ * Inspired by the code from https://github.com/soatok/minisign-php by Soatok Dreamseeker
+ */
+class SignatureHelper
+{
+    public const UNTRUSTED_COMMENT_PREFIX = 'untrusted comment: ';
+
+    public const REGEX = '#^' . self::UNTRUSTED_COMMENT_PREFIX . '(.+?)[\r\n\s]+([A-Za-z0-9+/=]+)[\s]+?$#';
```

Comment:
Maybe rename to `KEY_REGEX` or give it any name that describes what it does.

To me it looks like there **must** be a new line in between the comment and the base64-encoded key ([secret-key-format](https://jedisct1.github.io/minisign/#secret-key-format))? However, with this regex it could be any whitespace character except of a newline. Also, `\r\n` is redundant to `\s`.
Not sure if there **must** be whitespace after the base64-encoded key.
Why do you store the first group but then don't use it anywhere? Could be a non capturing group `(?:.+?)`.

Perhaps like so: `'#^' . self::UNTRUSTED_COMMENT_PREFIX . '(?:.+?)(?:\r\n?|\n)([A-Za-z0-9+/=]+)\s*?$#`

Reply:
I took the same regex as the reference php implementation, didn't look more into it TBH. It works and that's what matter. We can revisit it later. I've renamed it though, I agree that `REGEX` is very poor variable name.